compiler/luci/pass/src/CircleOptimizer.cpp

```diff
@@ -109,6 +109,12 @@ void CircleOptimizer::optimize(loco::Graph *g) const
   logo::Phase phase;
 
   /* TRANSFORM DECLARATION BEGIN */
+
+  // Shape inference is needed for added nodes doing below transformations
+  phase.emplace_back(std::make_unique<luci::ShapeInferencePass>());
+  phase.emplace_back(std::make_unique<luci::TypeInferencePass>());
+  phase.emplace_back(std::make_unique<logo::RemoveDeadNodeWithQueryPass>());
```

Comment:
Some passes create dead nodes (ex: FoldDequantizePass), so this pass should be called after such a pass is run.

Reply:
If  other passes create dead nodes, those nodes will be removed after the iteration.
However, considering the efficiency, I will apply your opinion :)

Reply:
> those nodes will be removed after the iteration.

AFAIK, a phase runner repeats to run passes only when a pass changes a graph (returns changed=true). FoldDequantizePass can return (changed = false) even though it creates a dead node.

https://github.com/Samsung/ONE/pull/4404#discussion_r494253979

Reply:
Oh..... I did another mistake on `FoldDequantizePass`... if any changes occurred in some pass, it should return true but I didn't..... really thanks for that.